tests/materialize/run.sh

```diff
@@ -2,116 +2,134 @@
 # 1. the names of prepared flow collections.
 export TEST_COLLECTION_SIMPLE="tests/${CONNECTOR}/simple"
+export TEST_COLLECTION_DUPLICATED_KEYS="tests/${CONNECTOR}/duplicated-keys"
 export TEST_COLLECTION_MULTIPLE_DATATYPES="tests/${CONNECTOR}/multiple-datatypes"
 
-# 2. the local paths of test datasets that matches the test collections.
+# 2. the paths of test datasets that matches the test collections.
 # - a dataset that matches the schema of TEST_COLLECTION_SIMPLE.
-export DATASET_SIMPLE="tests/materialize/datasets/simple.jsonl"
+export DATASET_SIMPLE="${ROOT_DIR}/tests/materialize/datasets/simple.jsonl"
+# - a dataset that matches the schema of TEST_COLLECTION_DUPLICATED_KEYS.
+export DATASET_DUPLICATED_KEYS="${ROOT_DIR}/tests/materialize/datasets/duplicated-keys.jsonl"
 # - a dataset that matches the schema of TEST_COLLECTION_MULTIPLE_DATATYPES.
-export DATASET_MULTIPLE_DATATYPES="tests/materialize/datasets/multiple-datatypes.jsonl"
-
-# 3. websocket URLs for ingesting data.
-WEBSOCKET_URL_PREFIX=ws://localhost:8080/ingest
-# URL for ingesting data to TEST_COLLECTION_SIMPLE.
-export WEBSOCKET_URL_SIMPLE=${WEBSOCKET_URL_PREFIX}/${TEST_COLLECTION_SIMPLE}
-# URL for ingesting data to TEST_COLLECTION_MULTIPLE_DATATYPES.
-export WEBSOCKET_URL_MULTIPLE_DATATYPES=${WEBSOCKET_URL_PREFIX}/${TEST_COLLECTION_MULTIPLE_DATATYPES}
+export DATASET_MULTIPLE_DATATYPES="${ROOT_DIR}/tests/materialize/datasets/multiple-datatypes.jsonl"
+
+# 3. the binding number of datasets in the push capture.
+export BINDING_NUM_SIMPLE=0
+export BINDING_NUM_DUPLICATED_KEYS=1
+export BINDING_NUM_MULTIPLE_DATATYPES=2
+
+# Util function to run flowctl command via docker.
+function runFlowctl() {
+    local script="$1"
+    local args=$2
+    local detached
+    if [[ "$3" == true ]]; then
+      detached="-d"
+    fi
 
-# Start the catalog for testing.
-function startFlow() {
+    local test_dir_target=/home/flow/tests/temp
+    local test_scripts_dir_target=/home/flow/scripts
     # Run as root, not the flow user, since the user within the container needs to access the
     # docker socket.
-    docker run  -d --rm -it \
-        --name "${FLOW_CONTAINER_NAME}" \
+
+    docker run ${detached:+"$detached"} \
+        --name "${FLOW_CONTAINER_NAME_PREFIX}-${script}-${EPOCHREALTIME}" \
         --user 0  \
-        --mount type=bind,source=/tmp,target=/tmp \
         --mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock \
-        --mount type=bind,source="$(pwd)",target=/home/flow/project \
+        --mount type=bind,source="${TEST_SCRIPTS_DIR}",target=${test_scripts_dir_target} \
+        --mount type=bind,source="${TEST_DIR}",target=${test_dir_target} \
+        --env BROKER_ADDRESS="http://${BROKER_ADDRESS}" \
+        --env CONSUMER_ADDRESS="http://${CONSUMER_ADDRESS}" \
+        --env TEST_DIR="${test_dir_target}" \
+        --env BUILDS_ROOT="file://${test_dir_target}/build/" \
+        --env BUILD_ID=run-test-"${CONNECTOR}" \
+        --env CATALOG \
```

Comment:
What is the value of `CATALOG` here? Is that taking the value that's currently set in the environment?

Reply:
Yes, exactly. This is an abbreviation of `--env CATALOG=${CATALOG}`, that docker run recognize.